libyara/proc.c

```diff
@@ -386,6 +387,22 @@ int _yr_process_attach(
     return ERROR_COULD_NOT_ATTACH_TO_PROCESS;
   }
 
+  status = 0;
+  if (waitpid(-1, &status, 0) == -1)
```

Comment:
Shouldn't this be `waitpid(pid, &status, 0)` ?

There could be multiple threads using `yr_proc_scan_proc` concurrently and an application using _libyara_ may have forked other processes.